### ISSUE TITLE
BC PR A: Demonstrations dataset + BcConfig

### ISSUE THREAD

**Issue by rjwalters** [rjwalters/thrust#164]
**Part of #161** (BC epic). **No unmet dependencies — immediately buildable.**

## Background
Behavioral Cloning (BC) trains a policy to reproduce expert actions from a fixed dataset of `(observation, action)` pairs via supervised cross-entropy. Before the trainer (#163, PR B) can exist, it needs (1) a dataset container to hold the demonstrations and serve shuffled minibatches, and (2) a config struct for the supervised hyperparameters. This is PR A of the BC epic.

Per the epic decision (#161, decision 2), the demo dataset is a small NEW struct, NOT a reuse of `RolloutBuffer`/`ReplayBuffer` (those carry rewards/dones/advantages/priorities that BC has no use for). Per decision 3, `BcConfig` mirrors the `A2cConfig`/`SacConfig` builder + `validate()` convention.

### 1. `Demonstrations` dataset (place in `src/train/bc/dataset.rs`)
A flat supervised dataset of discrete-action demonstrations.
```rust
pub struct Demonstrations { /* obs: Vec<f32>, actions: Vec<i64>, obs_dim, len */ }
impl Demonstrations {
    pub fn new(obs: Vec<f32>, actions: Vec<i64>, obs_dim: usize) -> anyhow::Result<Self>;
    pub fn len(&self) -> usize;
    pub fn is_empty(&self) -> bool;
    pub fn obs_dim(&self) -> usize;
    /// Gather the given example indices into (obs [k, obs_dim], actions [k]) tensors.
    pub fn batch<B: Backend>(&self, indices: &[usize], device: &B::Device)
        -> (Tensor<B, 2>, Tensor<B, 1, Int>);
}
```
- `new` validates `obs.len() == len * obs_dim` and `actions.len() == len`, returns `Err` otherwise.
- `batch` gathers a host-side minibatch by index and builds the tensors (no gradient concerns — these are inputs/labels).

### 2. `BcConfig` (place in `src/train/bc/config.rs`)
Mirror `src/train/a2c/config.rs` exactly (builder setters + `validate()` + `Default` + `new()`).

| Knob | Default |
|---|---|
| `learning_rate` | `1e-3` |
| `batch_size` | `64` |
| `epochs` | `10` |
| `seed` | `0` |

`validate()`: `learning_rate > 0`, `batch_size > 0`, `epochs > 0`.

### 3. Module wiring
Create `src/train/bc/mod.rs` declaring `pub mod config;` and `pub mod dataset;` with `pub use` re-exports (`BcConfig`, `Demonstrations`). Add `pub mod bc;` + re-exports to `src/train/mod.rs`.

## Acceptance criteria
- `Demonstrations::new` validates dimensions and returns `Err` on mismatch.
- `Demonstrations::batch` returns correctly-shaped `(Tensor<B,2>, Tensor<B,1,Int>)` for a given index slice (unit test on `Autodiff<NdArray<f32>>`).
- `BcConfig` has builder setters, `Default`, `new()`, and `validate()` rejecting non-positive `learning_rate`/`batch_size`/`epochs` (unit tests mirroring `a2c/config.rs` tests).
- `src/train/bc/{mod,config,dataset}.rs` compile under `--features training`; `bc` is re-exported from `src/train/mod.rs`.
- `cargo test --features training` passes the new unit tests.

## Affected files
- NEW `src/train/bc/mod.rs`, `src/train/bc/config.rs`, `src/train/bc/dataset.rs`
- EDIT `src/train/mod.rs` (add `pub mod bc;` + `pub use bc::{BcConfig, Demonstrations};`)

## References
- Config pattern: `src/train/a2c/config.rs` (builder + `validate()`).
- Tensor `gather`/construction: `src/policy/mlp.rs` (`evaluate_actions`, `TensorData::new`).
- Existing buffers (for contrast — do NOT reuse): `src/buffer/rollout.rs`, `src/buffer/replay.rs`.
